binder/src/androidTest/java/io/grpc/binder/internal/LeakSafeOneWayBinderTest.java

```diff
@@ -71,7 +78,21 @@ public void testDetach() {
     assertThat(binder.onTransact(456, p, null, FLAG_ONEWAY)).isFalse();
 
     // The transaction shouldn't have been processed.
-    assertThat(transactionsHandled).isEqualTo(0);
+    assertThat(handler.transactionsHandled).isEqualTo(0);
+
+    p.recycle();
+  }
+
+  @Test
+  public void testReplace() {
+    binder = new LeakSafeOneWayBinder(handler);
+    Parcel p = Parcel.obtain();
+    FakeHandler handler2 = new FakeHandler();
+    binder.setHandler(handler2);
+    assertThat(binder.onTransact(456, p, null, FLAG_ONEWAY)).isFalse();
+
+    assertThat(handler.transactionsHandled).isEqualTo(0);
+    assertThat(handler2.transactionsHandled).isEqualTo(0);
```

Comment:
I'm a bit confused by this. Shouldn't `handler2` have recorded the transaction?

Or maybe the test name is missing clarification about what the expected outcome here is.

Reply:
Yes sorry this test case was failing but I didn't notice because the `connectedCheck` don't run automatically as presubmits and there was a preexisting failure in that suite at head. Fixed.